### PR TITLE
Remove duplicates on dqn, ddpg, sac

### PR DESCRIPTION
**Pull request by Curt-Park**

### CODE REVIEW

**Comment by Curt-Park** [comment]
@MrSyee 앞으로 확장 모델을 작성시 상속구조를 활용하여 initialize 함수 및 update 함수만 작성하면 됩니다.

File: algorithms/dqn/agent.py [hunk 1]
```diff
@@ -144,29 +153,37 @@ def update_model(self, experiences: Tuple[torch.Tensor, ...]) -> torch.Tensor:
...
         #       = r                       otherwise
         masks = 1 - dones
-
         target = rewards + self.hyper_params["GAMMA"] * next_q_value * masks
         target = target.to(device)
 
-        loss = torch.mean((target - curr_q_value).pow(2) * weights)
-        # regularization
-        loss += torch.norm(q_values, 2).mean() * self.hyper_params["W_Q_REG"]
+        # calculatte dq loss
```
Comment (MrSyee) [inline algorithms/dqn/agent.py]: should modify this typo.

File: algorithms/fd/dqn_agent.py [hunk 2]
```diff
@@ -240,6 +138,7 @@ def write_log(self, i: int, avg_loss: dict, score: int = 0):
                 avg_loss[0],
                 avg_loss[1],
                 avg_loss[2],
+                datetime.datetime.now(),
```
Comment (MrSyee) [inline algorithms/fd/dqn_agent.py]: Why do you log datetime on dqn?

File: README.md [hunk 3]
```diff
@@ -27,7 +27,7 @@ We are warmly welcoming external contributors! :)
 3. [Proximal Policy Optimization Algorithms (PPO)](https://github.com/medipixel/rl_algorithms/blob/master/algorithms/ppo)
 4. [Twin Delayed Deep Deterministic Policy Gradient Algorithm (TD3)](https://github.com/medipixel/rl_algorithms/blob/master/algorithms/td3)
 5. [Soft Actor Critic Algorithm (SAC)](https://github.com/medipixel/rl_algorithms/blob/master/algorithms/sac/agent.py)
-6. [Behaviour Cloning (BC with DDPG)](https://github.com/medipixel/rl_algorithms/tree/master/algorithms/bc)
+6. [Behaviour Cloning (BC with DDPG, SAC)](https://github.com/medipixel/rl_algorithms/tree/master/algorithms/bc)
 7. [Prioritized Experience Replay (PER with DDPG)](https://github.com/medipixel/rl_algorithms/tree/master/algorithms/per)
 8. [From Demonstrations (DDPGfD, SACfD, DQfD)](https://github.com/medipixel/rl_algorithms/tree/master/algorithms/fd)
 9. [Deep Q-Network (DDQN)](https://github.com/medipixel/rl_algorithms/tree/master/algorithms/dqn)
```
Comment (MrSyee) [inline README.md]: should add dueling.

File: algorithms/dqn/agent.py [hunk 4]
```diff
@@ -235,6 +260,7 @@ def train(self):
             score += reward[0]
             i_episode_prev = i_episode
             i_episode += done.sum()
+            self.i_episode = i_episode
 
             if (i_episode // self.args.save_period) != (
```
Comment (MrSyee) [inline algorithms/dqn/agent.py]: should change "!=" to ">" ?

Comment (Curt-Park) [reply algorithms/dqn/agent.py]: doesn't really matter unless i_episode decreases

File: algorithms/dqn/agent.py (hunk 1 above)
Comment (Curt-Park) [reply algorithms/dqn/agent.py]: I will fix it

File: algorithms/fd/dqn_agent.py (hunk 2 above)
Comment (Curt-Park) [reply algorithms/fd/dqn_agent.py]: That is to know how much time flies. It prints out the current time on log.

File: README.md (hunk 3 above)
Comment (Curt-Park) [reply README.md]: I will change DDQN -> Dueling DQN

**Pull request merged**